src/audiomixerboard.cpp

```diff
@@ -1335,12 +1335,16 @@ void CAudioMixerBoard::ApplyNewConClientList ( CVector<CChannelInfo>& vecChanInf
 
 void CAudioMixerBoard::SetFaderLevel ( const int iChannelIdx, const int iValue )
 {
+    // Proposed change: if iChannelIdx is -1 and our own channel ID is a valid index
+    // then we adjust our own fader level:
+    const int iTheChannelIdx = ( iChannelIdx == INVALID_INDEX ) ? iMyChannelID : iChannelIdx;
```

Comment:
I don't like this implementation. Why abuse invalid index?

Reply:
Not sure if you need my involvement at this stage but my patch used `-1` explicitly at both the `emit` statement and here at `SetFaderLevel`; at first I didn't even know that is was the same value as INVALID_INDEX. I can definitely see your point.

Reply:
Yes.  And `INVALID_INDEX` is used to mean "the appropriate channel" in other places, too, already.  The only problem with `-1` for me is that it wasn't clear what it meant, whereas `INVALID_INDEX` in both places means "magic out of range value" quite clearly.

Reply:
Well, if one knows about this magic double meaning of "invalid", fine. I'm happy to leave it to you guys to decide how it's best done, I'm just happy you are considering adding this feature at all. Much appreciated.

Reply:
I'd rather use something else. Abusing invalid index doesn't feel right at all.

Reply:
We'll need to go through the whole application finding where it's been used like this, then.

Alternatively, I could bump all the numbers up, so they're "1-based" and reserve 0 for "my channel", then subtract one...

Reply:
Semantically speaking, it's just not INVALID_INDEX in my opinion.

Can we introduce another DEFINE saying something better than INVALID_INDEX? it could still be -1 internally.

Reply:
OK, changed to `I_MY_CHANNEL`.